tree/dataframe/src/RDataFrame.cxx

```diff
@@ -1562,15 +1631,37 @@ RDataFrame::RDataFrame(std::string_view treeName, TDirectory *dirPtr, const Colu
 /// The default columns are looked at in case no column is specified in the
 /// booking of actions or transformations.
 /// \see ROOT::RDF::RInterface for the documentation of the methods available.
-RDataFrame::RDataFrame(std::string_view treeName, std::string_view filenameglob, const ColumnNames_t &defaultColumns)
-   : RInterface(std::make_shared<RDFDetail::RLoopManager>(nullptr, defaultColumns))
+#ifdef R__HAS_ROOT7
+RDataFrame::RDataFrame(std::string_view treeName, std::string_view fileNameGlob, const ColumnNames_t &defaultColumns)
+   : RInterface(CreateLMFromFile(treeName, fileNameGlob, defaultColumns))
+{
+}
+
+/// \cond HIDDEN_SYMBOLS
+/// Private constructor, only used by the Snapshot operation
+RDataFrame::RDataFrame(int /*tagForTTree*/, std::string_view datasetName, std::string_view fileNameGlob,
+                       const ColumnNames_t &defaultColumns)
+   : RInterface(CreateLMFromTTree(datasetName, fileNameGlob, defaultColumns))
+{
+}
+
+/// The function is declared in RInterface.hxx and defined here. This allows
+/// creating `shared_ptr<ROOT::RDataFrame>` using the private constructor in
+/// in functions that are defined in RInterface.hxx, even though RDataFrame.hxx
+/// depends from RInterface.hxx.
+std::shared_ptr<ROOT::RDataFrame>
+ROOT::Internal::RDF::CreateRDFFromTTree(int tagForTTree, std::string_view treeName, std::string_view fileNameGlob,
+                                        const ROOT::RDF::ColumnNames_t &defaultColumns)
+{
+   return std::shared_ptr<ROOT::RDataFrame>(new ROOT::RDataFrame(tagForTTree, treeName, fileNameGlob, defaultColumns));
```

Comment:
consider using `make_shared` instead (shortens the line and avoids the raw `new`)